modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/SubtypesModelValidator.java

```diff
@@ -44,71 +47,89 @@ public void validate(ExtensionModel model) throws IllegalModelDefinitionExceptio
     Optional<Map<MetadataType, List<MetadataType>>> typesMapping =
         model.getModelProperty(SubTypesModelProperty.class).map(SubTypesModelProperty::getSubTypesMapping);
     if (typesMapping.isPresent()) {
+      validateBaseTypeNotFinal(model, typesMapping.get());
       validateNonAbstractSubtypes(model, typesMapping.get());
+      validateSubtypesSupportGlobalDeclaration(model, typesMapping.get());
       validateSubtypesExtendOrImplementBaseType(model, typesMapping.get());
       validateSubtypesNameClashing(model, typesMapping.get());
-      validateBaseTypeNotFinal(model, typesMapping.get());
     }
   }
 
+  private void validateSubtypesSupportGlobalDeclaration(ExtensionModel model,
+                                                        Map<MetadataType, List<MetadataType>> typesMapping) {
+    List<MetadataType> nonGlobalTypes = typesMapping.values().stream().flatMap(Collection::stream)
+        .filter(s -> s.getAnnotation(NotGlobalTypeAnnotation.class).isPresent()).collect(toList());
+
+    if (!nonGlobalTypes.isEmpty()) {
+      throw new IllegalModelDefinitionException(format("All the declared SubtypesMapping in extension %s must support to be declared"
+          + " as global types, but [%s] do not support it",
+                                                       model.getName(), Arrays.toString(nonGlobalTypes.toArray())));
```

Comment:
Just doing a `toString()` for a `MetadataType` will generate an unreadable StackTrace